sycl/CMakeLists.txt

```diff
@@ -220,28 +220,25 @@ if (MSVC)
   list(APPEND SYCL_RT_LIBS sycld)
 endif()
 
-function(add_libcxx_compile_options LIB_NAME)
-  target_compile_options(${LIB_NAME} PRIVATE -nostdinc++)
-  target_include_directories(${LIB_NAME} PRIVATE "${SYCL_LIBCXX_INCLUDE_PATH}")
-endfunction(add_libcxx_include_option)
-
-function(add_libcxx_link_options LIB_NAME)
-  target_link_libraries(${LIB_NAME} PRIVATE "-L${SYCL_LIBCXX_LIBRARY_PATH}" -Wl,-rpath,${SYCL_LIBCXX_LIBRARY_PATH} -nodefaultlibs -lc++ -lc++abi -lm -lc -lgcc_s -lgcc)
-endfunction(add_libcxx_link_options)
-
 # This function allows building multiple libraries with the same options.
 # Currently used by sycl and plugins library.
 # Currently handles linking with libcxx support and gcc workaround
-function( add_common_options LIB_NAME)
+# If IS_LINK is 1, common linker options should be added for target or common
+# compile option should be added.
+function( add_common_options TAR_NAME IS_LINK)
```

Comment:
I think using optional parameter for setting Object library name is much cleaner interface.

Just check if additional argument is set via `ARGC` variable and if so, set compile options for Object library and link options for the first argument. Keep current behavior otherwise. This way you don't need to change multiple CMakeLitsts.txt files - just sycl/source/CMakeLists.txt.

``` CMake
add_common_options(${LIB_NAME} ${LIB_OBJ_NAME})
```

See https://cmake.org/cmake/help/latest/command/function.html#arguments for more details.